dhcpv4/modifiers.go

```diff
@@ -43,6 +43,18 @@ func WithGatewayIP(ip net.IP) Modifier {
 	}
 }
 
+//WithRelayOptions copies the relay options from the request to the reply
+func WithRelayOptions(request *DHCPv4) Modifier {
+	return func(d *DHCPv4) {
+		// If request has Relay Agent Info copy it to the reply
+		if request.Options.Has(OptionRelayAgentInformation) {
```

Comment:
I think this can be simplified:

```
	if relayOpt := request.RelayAgentInfo(); relayOpt != nil {
		d.UpdateOption(dhcpv4.Option{Code: dhcpv4.OptionRelayAgentInformation, Value: relayOpt})
	}
```

Reply:
Looks good to me.  What do you think about the func name WithRelayAgentInfo?